porch/apiserver/pkg/registry/porch/packagerevision.go

```diff
@@ -269,3 +202,55 @@ func (r *packageRevisions) Delete(ctx context.Context, name string, deleteValida
 	// TODO: Should we do an async delete?
 	return oldObj, true, nil
 }
+
+// PackageRevisions Update Strategy
+
+type packageRevisionStrategy struct{}
+
+var _ SimpleRESTUpdateStrategy = packageRevisionStrategy{}
+
+func (s packageRevisionStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+}
+
+func (s packageRevisionStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	allErrs := field.ErrorList{}
+	oldRevision := old.(*api.PackageRevision)
+	newRevision := obj.(*api.PackageRevision)
+
+	switch lifecycle := oldRevision.Spec.Lifecycle; lifecycle {
+	case "", api.PackageRevisionLifecycleDraft, api.PackageRevisionLifecycleProposed:
+		// valid
+
+	default:
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "lifecycle"), lifecycle, fmt.Sprintf("can only update package with lifecycle value one of %s",
+			strings.Join([]string{
```

Comment:
We do have generics now ;-)

Reply:
not yet :( we are still building with 1.17 :)
https://github.com/GoogleContainerTools/kpt/blob/main/.github/workflows/porch.yml#L35

but we can now upgrade :)